google-cloud-storage/src/main/java/com/google/cloud/storage/Storage.java

```diff
@@ -1255,6 +1255,55 @@ public static SignUrlOption withQueryParams(Map<String, String> queryParams) {
     }
   }
 
+  /** Class for specifying watchAll options. */
+  class WatchAllOption extends Option {
+
+    private WatchAllOption(StorageRpc.Option rpcOption, Object value) {
+      super(rpcOption, value);
+    }
+
+    /** Returns an option to set a delimiter. */
+    public static WatchAllOption delimiter(String delimiter) {
+      return new WatchAllOption(StorageRpc.Option.DELIMITER, delimiter);
+    }
+
+    /** Returns an option to specify the maximum number of objects returned per page. */
+    public static WatchAllOption maxResults(Integer maxResults) {
+      return new WatchAllOption(StorageRpc.Option.MAX_RESULTS, maxResults);
+    }
+
+    /** Returns an option to specify the page token from which to start listing objects. */
+    public static WatchAllOption pageToken(String pageToken) {
+      return new WatchAllOption(StorageRpc.Option.PAGE_TOKEN, pageToken);
+    }
+
+    /**
+     * Returns an option to set a prefix to filter results to objects whose names begin with this
+     * prefix.
+     */
+    public static WatchAllOption prefix(String prefix) {
+      return new WatchAllOption(StorageRpc.Option.PREFIX, prefix);
+    }
+
+    /**
+     * Returns an option to define the projection in the API request. In some cases this option may
+     * be needed to be set to `noAcl` to omit ACL data from the response. The default value is
+     * `full`.
+     */
+    public static WatchAllOption projection(String projection) {
+      return new WatchAllOption(StorageRpc.Option.PROJECTION, projection);
+    }
+
+    /**
+     * If set to {@code true}, lists all versions of an object. The default is {@code false}.
```

Comment:
I think only `lists all versions of an object. The default is {@code false}. ' is enough,

Reply:
i think we need to provide link here as well for more info

Reply:
I am sorry for confusion, Link is ok I meant remove first part of the sentence i.e.

`~~If set to {@code true},~~ lists all versions of an object. The default is {@code false}.

Reply:
`If set to {@code true}`, indicates that what would be returns if its true